File: lib/cache/license/license_use.dart
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

import 'license_usecase.dart';

class LicenseUse {
  List<LicenseUsecase> usecases;
  List<String>? destinations;

  LicenseUse(this.usecases, {this.destinations});

  LicenseUse.fromMap(Map<String, dynamic> map)
      : usecases = map["usecases"]
            .map<LicenseUsecase>((usecase) => LicenseUsecase.from(usecase))
            .toList(),
        destinations = map["destinations"] != null
            ? List<String>.from(map["destinations"])
            : null;

  Map toMap() => {
        "usecases": usecases.map((usecase) => usecase.value).toList(),
        "destinations": destinations
      };
}

File: src/handler/license_use.rs
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

use serde::{Deserialize, Serialize};
use super::UseCase;

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct LicenseUse {
    #[serde(alias = "usecases")]
    use_cases: Vec<UseCase>,
    destinations: Option<Vec<String>>
}

impl LicenseUse {
    pub fn use_cases(&self) -> &Vec<UseCase> { &self.use_cases }
    pub fn destinations(&self) -> &Option<Vec<String>> { &self.destinations }
}
